ast.ts

```diff
@@ -100,7 +100,14 @@ export type Expr<A> =
   | { a?: A; tag: "method-call"; obj: Expr<A>; method: string; arguments: Array<Expr<A>> }
   | { a?: A; tag: "construct"; name: string }
   | { a?: A; tag: "lambda"; args: Array<string>; ret: Expr<A> }
-  | { a?: A; tag: "comprehension"; expr: Expr<A>; field: string; iter: Expr<A>; cond?: Expr<A> }
+  | {
+      a?: A;
+      tag: "comprehension";
+      expr: Expr<A>;
+      field: Assignable<A>;
+      iter: Expr<A>;
+      cond?: Expr<A>;
+    } // Need to change field to Assignable since lst = [t.n for t.n in range(10)] runs correctly, if the type of t has n.
```

Comment:
It might be worth using `Destructure<A>` (or `tag: "assignment"`) over `Assignable<A>` since python allows destructuring in this position (e.g. `[x for x, _ in [(1, 2)]]`), in particular because the `(tc|traverse|codeGen)Destructure` are the endpoints we plan to support for general purpose assignment. See #37 for proposed [API](https://github.com/ucsd-cse231-w21/chocopy-wasm-compiler/pull/37/files#diff-0a03798f0e47b1080516ea97c55cfc6c2e025276ef50bf1e1ede9b6e4570a33cR225-R238).

Reply:
This is a great suggestion @R167 . 

We were thinking of putting in `Destructure<A>` once the proposed API is merged, which doesn't seem to be the case right now. But assuming this gets merged before the next milestone, we are planning to make the change that you suggest.